atc/exec/get_step.go

```diff
@@ -191,6 +192,29 @@ func (step *GetStep) run(ctx context.Context, state RunState, delegate GetDelega
 		return false, err
 	}
 
+	getResult, found, err := step.getFromLocalCache(ctx, logger, step.metadata.TeamID, resourceCache, workerSpec)
+	if found {
+		fmt.Fprintln(delegate.Stderr(), "\x1b[1;36mINFO: found resouce cache from local cache\x1b[0m")
+		fmt.Fprintln(delegate.Stderr(), "")
+
+		state.StoreResult(step.planID, resourceCache)
+
+		state.ArtifactRepository().RegisterArtifact(
+			build.ArtifactName(step.plan.Name),
+			getResult.GetArtifact,
+		)
+
+		delegate.Finished(
+			logger,
+			ExitStatus(getResult.ExitStatus),
+			getResult.VersionResult,
+		)
+
```

Comment:
Do we want to update the version metadata like we do below?

```
		if step.plan.Resource != "" {
			delegate.UpdateVersion(logger, step.plan, getResult.VersionResult)
		}
```

I think this basically updates the metadata on the version that belongs to a `resource_config_scope_id`. There can be multiple `resource_config_scope_id`s that point to a single `resource_config_id`, which we probably want to update the metadata for the `resource_config_scope_id` that this resource points to even if it used a resource cache from the local cache.

Reply:
Make sense. Fixed.